lib/skate/detours/trip_modification.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Skate.Detours.TripModification do
+  @moduledoc """
+  A module that produces GTFS-flavored TripModification structs for a detour
+
+  More info: https://gtfs.org/realtime/reference/#message-tripmodifications
+  """
+
+  alias Schedule.Gtfs.RoutePattern
+  alias Schedule.Gtfs.Stop
+
+  defmodule Input do
+    @moduledoc """
+    Input for `Skate.Detours.TripModification.for/1`.
+    """
+
+    @type t :: %__MODULE__{
+            route_pattern: RoutePattern.t(),
+            missed_stops: [Stop.t()]
+          }
+
+    defstruct [:route_pattern, :missed_stops]
+  end
+
+  defmodule SelectedTrip do
+    @moduledoc """
+    Struct representing the `selected_trips` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-selectedtrips
+    """
+
+    @type t :: %__MODULE__{
+            trip_ids: [String.t()]
+          }
+
+    defstruct [:trip_ids]
+  end
+
+  defmodule Modification do
+    @moduledoc """
+    Struct representing the `modifications` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-modification
+    """
+
+    @type t :: %__MODULE__{
+            start_stop_selector: String.t(),
+            end_stop_selector: String.t()
+          }
+
+    defstruct [:start_stop_selector, :end_stop_selector]
+  end
+
+  @type t :: %__MODULE__{
+          selected_trips: [SelectedTrip.t()],
+          modifications: [Modification.t()]
+        }
+  defstruct [:selected_trips, :modifications]
+
```

Comment:
question: is there a reason why the required field [`service_dates`](https://gtfs.org/realtime/reference/#message-tripmodifications) is not in this struct currently?

I'm okay with transit data modifying it down the line, but it seems like we should be providing _something_ here since it's required.

Reply:
@firestack what would put in as a simple default? Just like... today's date to start?

Reply:
Today's date seems like a good default to me.

Reply:
[Added `service_dates`](https://github.com/mbta/skate/pull/2665/commits/0f750f523c6719bfd87faa5ededb365630c9250e)

Reply:
I made it an input 😇